llama-index-integrations/embeddings/llama-index-embeddings-solar/llama_index/embeddings/solar/base.py

```diff
@@ -0,0 +1,97 @@
+"""Solar embeddings file."""
+
+import json
+import requests
+from typing import Any, List, Optional
+
+from llama_index.core.base.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+)
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+
+DEFAULT_SOLAR_API_BASE = "https://api.upstage.ai/v1/solar/embeddings"
+DEFAULT_SOLAR_MODEL = "solar-1-mini-embedding-query"
+
+
+class SolarEmbedding(BaseEmbedding):
+    """Solar embeddings."""
+
+    api_key: str = Field(description="The Solar API key.")
+    api_base: str = Field(
+        default=DEFAULT_SOLAR_API_BASE,
+        description="The base URL for Solar embedding API.",
+    )
+
+    def __init__(
+        self,
+        model_name: str = DEFAULT_SOLAR_MODEL,
+        api_key: str = "",
```

Comment:
@Tokkiu if user doesn't pass the api_key explicitly, this makes the empty api_key so that raise 401 error in my environment
```suggestion
        api_key: str = None,
```